kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -273,6 +275,9 @@ class BatchJobSubmission(
         process.waitFor(applicationCheckInterval, TimeUnit.MILLISECONDS)
```

Comment:
so the key point is - in the current round, app state is NOT_FOUND because the submit stage exceeds the `kyuubi.engine.yarn.submit.timeout`, while during the `process.waitFor` period, submit success, then `process.isAlive` returns `false`, it has no chance to retrieve the app state from the cluster manager?

Reply:
Yes,

I think with `spark.yarn.submit.waitAppCompletion=false`, usually the application state is UNKNOWN/NOT_FOUND when process exit.

Reply:
When the process terminated, if the current running time not exceed `kyuubi.engine.yarn.submit.timeout`, the app state is UNKNOWN.

If it exceeds the engine submit timeout, the app state would be `NOT_FOUND`.